apis/core/v1beta1/function_webhook.go

```diff
@@ -17,11 +17,165 @@ limitations under the License.
 package v1beta1
 
 import (
+	"fmt"
+	"reflect"
+	"regexp"
+
+	shipwrightv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"k8s.io/api/autoscaling/v2beta2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+var (
+	shipwrightStrategies = map[shipwrightv1alpha1.BuildStrategyKind]bool{
+		shipwrightv1alpha1.NamespacedBuildStrategyKind: true,
+		shipwrightv1alpha1.ClusterBuildStrategyKind:    true}
+	shipwrightStrategiesSlice      = convertMapKeysToStringSlice(shipwrightStrategies)
+	funcRuntimes                   = map[Runtime]bool{Knative: true, Async: true}
+	funcRuntimesSlice              = convertMapKeysToStringSlice(funcRuntimes)
+	scaledObjectWorkloadTypes      = map[string]bool{"Deployment": true, "StatefulSet": true}
+	scaledObjectWorkloadTypesSlice = convertMapKeysToStringSlice(scaledObjectWorkloadTypes)
+	scaleJobRestartPolices         = map[v1.RestartPolicy]bool{
+		v1.RestartPolicyAlways:    true,
+		v1.RestartPolicyOnFailure: true,
+		v1.RestartPolicyNever:     true,
+	}
+	scaleJobRestartPolicesSlice = convertMapKeysToStringSlice(scaleJobRestartPolices)
+	pubsubComponentTypes        = map[string]bool{
+		"pubsub.kafka":            true,
+		"pubsub.snssqs":           true,
+		"pubsub.azure.eventhubs":  true,
+		"pubsub.azure.servicebus": true,
+		"pubsub.gcp.pubsub":       true,
+		"pubsub.hazelcast":        true,
+		"pubsub.in-memory":        true,
```

Comment:
So we need to remove all component name validation as discussed in the meeting

Reply:
Yes, the changes related to the meeting discussion have been completed. I'm still looking at the e2e test failures and will submit them together once resolved

Reply:
@tpiperatgod what do you think?

Reply:
Oh, in @webup‘s nodejs sample, there is no scaled option specified and we might need to allow this.
When there is no scale options, the deployment with 1 replica is created.

Reply:
> the scaledobject returned by GenScaleOptions shouldn't be nil? @tpiperatgod

yes, it can be nil. This depends on whether [scaleOption](https://github.com/OpenFunction/OpenFunction/blob/d2db62f2a5/apis/events/v1alpha1/event.go#L233) is set in EventBus or EventSource.